packages/teleport/src/Discover/TestConnection/TestConnection.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { ButtonPrimary, Text, Box, LabelInput, Flex } from 'design';
+import * as Icons from 'design/Icon';
+import Select from 'shared/components/Select';
+
+import { Header, ActionButtons, TextIcon } from '../Shared';
+import { useDiscoverContext } from '../discoverContextProvider';
+
+import { useTestConnection, State } from './useTestConnection';
+
+import type { Option } from 'shared/components/Select';
+import type { AgentStepProps } from '../types';
+
+export default function Container(props: AgentStepProps) {
+  const ctx = useDiscoverContext();
+  const state = useTestConnection({ ctx, props });
+
+  return <TestConnection {...state} />;
+}
+
+export function TestConnection({
+  attempt,
+  startSshSession,
+  logins,
+  runConnectionDiagnostic,
+  diagnosis,
+}: State) {
+  const usernameOpts = logins.map(l => ({ value: l, label: l }));
+  // There will always be one login, as the user cannot proceed
+  // the step that requires users to have at least one login.
+  const [selectedOpt, setSelectedOpt] = useState(usernameOpts[0]);
```

Comment:
This should probably go into a `useEffect` with a dep on `logins`.

Reply:
hmm, i don't see the benefit, or maybe i'm not understand why?

Reply:
hmm, i don't see the benefit, or maybe i'm not understand why?

Reply:
Oh it's just a micro perf improvement. Basically the map will run every time the component is re-rendered vs when the list of `logins` changes.